WooCommerce/Classes/POS/Presentation/TotalsView.swift

```diff
@@ -72,21 +33,80 @@ struct TotalsView: View {
+                Spacer().frame(height: Constants.subtotalsVerticalSpacing)
+                subtotalFieldView(title: Localization.taxes,
+                                  formattedPrice: totalsViewModel.formattedOrderTotalTaxPrice,
+                                  shimmeringActive: totalsViewModel.isShimmering,
+                                  redacted: totalsViewModel.isTaxFieldRedacted)
+                Spacer().frame(height: Constants.totalVerticalSpacing)
+                Divider()
+                    .overlay(Color.posTotalsSeparator)
+                Spacer().frame(height: Constants.totalVerticalSpacing)
+                totalFieldView(formattedPrice: totalsViewModel.formattedOrderTotalPrice,
+                               shimmeringActive: totalsViewModel.isShimmering,
+                               redacted: totalsViewModel.isTotalPriceFieldRedacted)
+            }
+            .padding(Constants.totalsLineViewPadding)
+            .frame(minWidth: Constants.pricesIdealWidth)
+            .fixedSize(horizontal: true, vertical: false)
+            Spacer()
         }
-        else {
-            return [
-                Gradient.Stop(color: Color.clear, location: 0.0),
-                Gradient.Stop(color: Color.posTotalsGradientPurple, location: 1.0)
-            ]
+    }
+
+    @ViewBuilder
+    func subtotalFieldView(title: String, formattedPrice: String?, shimmeringActive: Bool, redacted: Bool) -> some View {
+        if shimmeringActive {
+            shimmeringLineView(width: Constants.shimmeringWidth, height: Constants.subtotalsShimmeringHeight)
+        } else {
+            HStack(alignment: .top, spacing: .zero) {
+                Text(title)
+                    .font(Constants.subtotalTitleFont)
+                Spacer()
+                Text(formattedPrice ?? "")
+                    .font(Constants.subtotalAmountFont)
+                    .redacted(reason: redacted ? [.placeholder] : [])
+            }
+            .foregroundColor(Color.primaryText)
         }
     }
+
+    @ViewBuilder
+    func totalFieldView(formattedPrice: String?, shimmeringActive: Bool, redacted: Bool) -> some View {
+        if shimmeringActive {
+            shimmeringLineView(width: Constants.shimmeringWidth, height: Constants.totalShimmeringHeight)
+        } else {
+            HStack(alignment: .top, spacing: .zero) {
+                Text(Localization.total)
+                    .font(Constants.totalTitleFont)
+                    .fontWeight(.semibold)
+                Spacer(minLength: Constants.totalsHorizontalSpacing)
+                Text(formattedPrice ?? "")
+                    .font(Constants.totalAmountFont)
+                    .redacted(reason: redacted ? [.placeholder] : [])
+            }
+            .foregroundColor(Color.primaryText)
+        }
+    }
+
+    func shimmeringLineView(width: CGFloat, height: CGFloat) -> some View {
+        Color.posTotalsSeparator
```

Comment:
I need a whole line to be shimmering, rather than "Text", Space, "Text". Maybe you have more experience working with it an have any other ideas how to achieve it nicely?

Reply:
Since with the new design the whole "line" is shimmering I suppose this way of having a "blank" view/rectangle and shimmering it works ok!